### ISSUE TITLE
Change the SET default parameters to improve its accuracy

### ISSUE THREAD

**Issue by FedericoTartarini** [CenterForTheBuiltEnvironment/pythermalcomfort#108]
Melnikov found that the accuracy of the model can be improved by changing some of the default parameters in the SET model. In his paper (ref below) he suggests using the values in set 3 of the following table. In particular, in his analysis he suggest that changing the value of neutral core-skin blood flow from 6.3 to 10.7 has the most impact. Shall we update this value?

<img width="380" alt="image" src="https://github.com/CenterForTheBuiltEnvironment/pythermalcomfort/assets/40018640/cc97de6a-02e7-4702-9e04-4ffd9a54e06b">

Moreover, Melnikov also provides additional equations to calculate the convective heat transfer coefficient. See the screenshot below.

<img width="424" alt="image" src="https://github.com/CenterForTheBuiltEnvironment/pythermalcomfort/assets/40018640/70a01831-4a6d-457d-bc3e-5ee875ab9982">

I am tagging you in this discussion since I would like to know your feedback. We should also decide what it is the right procedure that we should follow when we suggest changes to the source code. I can open a separate discussion about this. @AkihisaNomoto @t-kramer @stefanoschiavon @eddes 

@article{melnikov_system_2018,
	title = {System dynamics of human body thermal regulation in outdoor environments},
	volume = {143},
	issn = {0360-1323},
	url = {https://www.sciencedirect.com/science/article/pii/S0360132318304323},
	doi = {10.1016/j.buildenv.2018.07.024},
	urldate = {2024-06-14},
	journal = {Building and Environment},
	author = {Melnikov, Valentin and Krzhizhanovskaya, Valeria V. and Lees, Michael H. and Sloot, Peter M. A.},
	month = oct,
	year = {2018},
	keywords = {Human body thermal regulation, Outdoor thermal comfort, System dynamics, Two-node model},
	pages = {760--769},
	file = {Accepted Version:/Users/ftar3919/Zotero/storage/3QX8AG2E/Melnikov et al. - 2018 - System dynamics of human body thermal regulation i.pdf:application/pdf;ScienceDirect Snapshot:/Users/ftar3919/Zotero/storage/QST4PK6H/S0360132318304323.html:text/html},
}

**Comment by AkihisaNomoto** [comment]
Hi @FedericoTartarini - Thank you for sharing the paper. It looks great. 

I took a look at the paper but was not sure where the author's contribution was and where the original parts were. I do not remember all the SET equations, but it seems like the author's contribution is what you showed in this issue (#108). 

**Basic setting of blood flow**: Honestly speaking, it is unclear whether the value of blood flow is true or not. Increasing the basic setting value of blood flow may make the core temperature response faster. The authors have tuned the model parameters to fit Munir's data, so it is natural that the accuracy is high for Munir's data. The important thing is whether the accuracy has been validated against other datasets. While the authors have validated against Fiala's data, this experiment does not capture the response when moving from a cold to a slightly warm environment, which is the issue that the author points out. Therefore, in my opinion, further accuracy validation is necessary (i.e., the original model might also reproduce Fiala's experiment well). If the authors can accurately reproduce experiments in a cold environment and improve the accuracy of skin and core temperatures, it would be worth incorporating to the code.

**Convective heat transfer coefficient**: I do not think this value has a significant impact, but the method of selecting the maximum value from five formulas does not make sense to me and this seems to break physical principles. These simplified formulas are originally determined by the shape and conditions of the object. It is unclear which literature the five formulas reference. Values like 3 and 8.6 are written as constants, but they should be expressed as symbols since they inherently have dimensions. Otherwise, the units on the right and left sides would be completely different.

**Source code modification**: To simplify code maintenance, you better avoid duplication. How about making the model type selectable based on the original model code? I know many researchers who have improved the SET model, but most of these improvements are parameter tuning rather than significant changes to the calculation logic, so I think it can be handled with simple code like the one below.
```
def set_tmp(args, model_type="original"):
    if model_type == "original":
        a = 1
        b = 2
    elif model_type == "pizza":
        a = 2
        b = 3
    elif model_type == "melnikov":
        a = 4
        b = 5
    else:
        print("Error: Model type should be one of ['original', 'melnikov', 'pizza']")
```

**Comment by ruijis** [comment]
FYI, I just read another paper about the accuracy of the SET model:
https://www.sciencedirect.com/science/article/pii/S0360132321011045

**Comment by eddes** [comment]
Hello all, 

Sorry for the late contribution to this topic, I hope it still may help.

I would be cautious with the parametrization of physiological parameters of the SET model developped by Gagge. In 
[this paper](https://www.researchgate.net/publication/380576458_The_transient_PET_comfort_index_A_Description_and_analysis_of_the_physiological_model), the 2-node model used for the SET is compared with Munir's data. On Figure 2. b), the results of the method used by Melnikov to best fit the skin temperature are shown (lines with "Gagge"). One can observe that the core temperature drops, whereas the experimental observation is an increase of the core temperature.

In other words, although the skin temperature's RMSE is lower than with the original values of the physiological parameters, the results on the core temperature do not fit as well and exhibit a behaviour that contradicts experimental observations (possibly a reason why the core temperature is not shown in Melnikov's paper?).

Regards,
